### PR TITLE
[JENKINS-34854] - Migrate missing settings to SystemProperties

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Follow-up to https://github.com/jenkinsci/jenkins/pull/2337
- [x] - Fix: Integer and Boolean settings have been forgotten
- [x] - Add: Long SystemProperties API
- [x] - Migration of Long settings to SystemProperties

I have not found properties of other type after the search.

https://issues.jenkins-ci.org/browse/JENKINS-34854

@reviewbybees

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
Also CC @jernst

**Comment by andresrc** [comment]
:bee:

File: core/src/main/java/jenkins/util/SystemProperties.java [hunk 1]
```diff
@@ -249,6 +249,51 @@ public static Integer getInteger(String name, Integer def) {
...
+    public static Long getLong(String name, Long def) {
+        String v = getString(name);
+       
+        if (v != null) {
+            try {
+                return Long.decode(v);
+            } catch (NumberFormatException e) {
+                // Ignore, fallback to default
+                if (LOGGER.isLoggable(Level.CONFIG)) {
+                    LOGGER.log(Level.CONFIG, "Property. Value is not integer: {0} => {1}", new Object[] {name, v});
```
Comment (rsandell) [inline core/src/main/java/jenkins/util/SystemProperties.java]: integer -> long

**Comment by rsandell** [comment]
:bee:

**Comment by armfergom** [comment]
LGTM

File: core/src/main/java/jenkins/util/SystemProperties.java (hunk 1 above)
Comment (oleg-nenashev) [inline core/src/main/java/jenkins/util/SystemProperties.java]: Yep, copy-pasta

**Comment by oleg-nenashev** [comment]
Fixed the minor comments.
CC @jenkinsci/code-reviewers

**Comment by andresrc** [comment]
:bee:

**Comment by armfergom** [comment]
🐝

**Comment by oleg-nenashev** [comment]
@reviewbybees done
@jenkinsci/code-reviewers WDYT?

**Pull request merged**